packages/polling/src/poll.ts

```diff
@@ -500,4 +515,44 @@ namespace Private {
 
     return Math.min(max, random);
   }
+
+  /**
+   * Keep track of whether the document is hidden. This flag is only relevant in
+   * a browser context.
+   *
+   * Listen to `visibilitychange` event to set the `hidden` flag.
+   *
+   * Listening to `pagehide` is also necessary because Safari support for
+   * `visibilitychange` events is partial, cf.
+   * https://developer.mozilla.org/docs/Web/API/Document/visibilitychange_event
+   */
+  export let hidden = (() => {
+    if (typeof document === 'undefined') {
+      return false;
+    }
+    document.addEventListener('visibilitychange', () => {
+      hidden = document.visibilityState === 'hidden';
+    });
+    document.addEventListener('pagehide', () => {
+      hidden = document.visibilityState === 'hidden';
+    });
+    return document.visibilityState === 'hidden';
+  })();
```

Comment:
To avoid external modification (aka hidden should not be modified except by the listeners), what about doing something like:

```suggestion
  let _hidden = (() => {
    if (typeof document === 'undefined') {
      return false;
    }
    document.addEventListener('visibilitychange', () => {
      _hidden = document.visibilityState === 'hidden';
    });
    document.addEventListener('pagehide', () => {
      _hidden = document.visibilityState === 'hidden';
    });
    return document.visibilityState === 'hidden';
  })();
  
  export function isHidden(): boolean {
    return _hidden;
  }
```

Reply:
I think that because `hidden` is already in the `Private` namespace, it is already protected from interference: it's true that something else in `poll.ts` could modify but nothing externally can change it.

It follows the same convention as other exported variables in private namespaces we use (e.g., in `@lumino/keyboard` and `@lumino/signaling`.

Reply:
Thanks for the comment